core/trino-main/src/main/java/io/trino/dispatcher/QueuedStatementResource.java

```diff
@@ -172,6 +174,10 @@ private Query registerQuery(String statement, HttpServletRequest servletRequest,
     {
         Optional<String> remoteAddress = Optional.ofNullable(servletRequest.getRemoteAddr());
         Optional<Identity> identity = Optional.ofNullable((Identity) servletRequest.getAttribute(AUTHENTICATED_IDENTITY));
+        if (identity.flatMap(Identity::getPrincipal).map(InternalPrincipal.class::isInstance).orElse(false)) {
```

Comment:
Should we change the meaning of `@ResourceSecurity AUTHENTICATED_USER` instead?

do we have resources that are both user-facing _and_ for internal consumption at the same time?
Assuming these are rare, we could just annotate them explicitly (eg `@ResourceSecurity({AUTHENTICATED_USER, INTERNAL})`)

Reply:
That is a much broader discussion.  Also, I'm not totally convinced that this what we want in the long run, but for not the internal communication is never used to start queries, so I think this is just extra safe for now.